tangos/live_calculation/query_masking.py

```diff
@@ -26,6 +26,24 @@ def mark_nones_as_masked(self, input):
         assert mask.shape==(self.N,)
         self.results_target = np.where(mask)
 
+    def mark_false_as_masked(self, input):
+        """Mark any rows in the input that are None as masked, excluding them from future queries."""
+        if self.N is None:
+            self.N = len(input)
+            to_mask = input
+        else:
+            to_mask = self.unmask(input)
+
+        if len(to_mask.shape)==2:
+            mask = np.any(is_not_false(to_mask),axis=0).reshape(self.N)
+        elif len(to_mask.shape)==1:
+            mask = is_not_false(to_mask)
+        else:
+            raise ValueError("Not able to use an input of this shape to determine a query mask")
+
+        assert mask.shape==(self.N,)
+        self.results_target = np.where(mask)
```

Comment:
This seems to be duplicating the exact same code from above. Maybe make a generic method like `_retain_row_if_input_satisfies(self, input, condition)` then `mark_false_as_masked` could call `_retain_row_if_input_satisfies(self, input, is_not_false)` and `mark_none_as_masked` could call `_retain_row_if_input_satisfies(self, input, is_not_none)`

Reply:
good idea I can do that!